http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala

```diff
@@ -237,17 +256,38 @@ class ProxyRoutes(proxyService: ProxyService, corsSupport: CorsSupport, refererC
     for {
       ctx <- ev.ask[AppContext]
       apiCall = proxyService.proxyRequest(userInfo, googleProject, runtimeName, request)
-      _ <- metrics.incrementCounter("proxyRuntime")
       resp <- ctx.span.fold(apiCall)(span => spanResource[IO](span, "proxyRuntime").use(_ => apiCall))
+      tool = if (request.uri.toString.contains("jupyter")) {
+        "Jupyter"
+      } else if (request.uri.toString.contains("rstudio")) {
+        "Rstudio"
+      } else if (request.uri.toString.contains("welder")) {
+        "Welder"
+      } else {
+        "other"
+      }
+
+      _ <- if (resp.status.isSuccess()) {
+        metrics.incrementCounter("proxyRequest",
+                                 tags = Map("result" -> "success", "action" -> "runtimeRequest", "tool" -> s"${tool}"))
+
+        if (request.uri.toString.endsWith(".ipynb") && request.method == HttpMethods.PUT) {
+          metrics.gauge("proxy/notebooksSize", resp.entity.getContentLengthOption().getAsLong.toDouble)
```

Comment:
I think `.getContentLengthOption().getAsLong` is the akka Java API, and probably throws an exception if the optional is not present.

I would do:
```
resp.entity.contentLengthOption.traverse { c =>
  metrics.gauge("proxy/notebooksSize", c.toDouble)
}
```
(I don't think it's too important to check the `.toDouble` conversion.)

Reply:
Also just realized this is using the `HttpResponse`. Do we want to do it on the `HttpRequest` instead? (Since I think the PUT _request_ would have the right content-length.)

Reply:
oops thank you yes should be the request